pkg/gardenlet/controller/seed/seed_extension_control.go

```diff
@@ -0,0 +1,152 @@
+		return err
+	}
+
+	return c.extensionCheckControl.ReconcileExtensionCheckFor(seed)
+}
+
+// ExtensionCheckControlInterface implements the control logic for updating Seeds. It is implemented as an interface to allow
+// for extensions that provide different semantics. Currently, there is only one implementation.
+type ExtensionCheckControlInterface interface {
+	// ReconcileExtensionCheckFor implements the control logic for Seed extension checks.
+	// If an implementation returns a non-nil error, the invocation will be retried using a rate-limited strategy.
+	// Implementors should sink any errors that they do not wish to trigger a retry, and they may feel free to
+	// exit exceptionally at any point provided they wish the update to be re-run at a later point in time.
+	ReconcileExtensionCheckFor(seed *gardencorev1alpha1.Seed) error
+}
+
+// NewDefaultExtensionCheckControl returns a new instance of the default implementation ExtensionCheckControlInterface that
+// implements the documented semantics for Seeds. You should use an instance returned from NewDefaultExtensionCheckControl() for any
+// scenario other than testing.
+func NewDefaultExtensionCheckControl(
+	k8sGardenClient kubernetes.Interface,
+	controllerInstallationLister gardencorelisters.ControllerInstallationLister,
+	recorder record.EventRecorder,
+) ExtensionCheckControlInterface {
+	return &defaultExtensionCheckControl{
+		k8sGardenClient,
+		controllerInstallationLister,
+		recorder,
+	}
+}
+
+type defaultExtensionCheckControl struct {
+	k8sGardenClient              kubernetes.Interface
+	controllerInstallationLister gardencorelisters.ControllerInstallationLister
+	recorder                     record.EventRecorder
+}
+
+func (c *defaultExtensionCheckControl) ReconcileExtensionCheckFor(obj *gardencorev1alpha1.Seed) error {
+	var (
+		seed                         = obj.DeepCopy()
+		conditionSeedExtensionsReady = gardencorev1alpha1helper.GetOrInitCondition(seed.Status.Conditions, gardencorev1alpha1.SeedExtensionsReady)
+	)
+
+	controllerInstallationList, err := c.controllerInstallationLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	message := ""
+
+	for _, controllerInstallation := range controllerInstallationList {
+		if controllerInstallation.Spec.SeedRef.Name != seed.Name {
+			continue
+		}
+
+		count := 0
+		for _, condition := range controllerInstallation.Status.Conditions {
+			if condition.Type != gardencorev1alpha1.ControllerInstallationValid && condition.Type != gardencorev1alpha1.ControllerInstallationInstalled && condition.Type != gardencorev1alpha1.ControllerInstallationHealthy {
+				continue
+			}
+
+			if condition.Status != gardencorev1alpha1.ConditionTrue {
+				message = fmt.Sprintf("Condition %q for extension %q is false: %s", condition.Type, controllerInstallation.Name, condition.Message)
+			} else {
+				count++
+			}
+		}
+
+		if count != 3 {
```

Comment:
instead of ifs: maybe declare a map map[string]struct{} for the valid condition.types (ControllerInstallationValid, ControllerInstallationInstalled, ControllerInstallationHealthy) and then check if condition.type is in this map.

Then you could also use the length of the map below and do not have to use the int 3. More maintainable & understandable.